{"priority": "trivial", "kind": "bug", "repository": {"links": {"self": {"href": "data/repositories/dedalus-project/dedalus.json"}, "html": {"href": "#!/dedalus-project/dedalus"}, "avatar": {"href": "data/bytebucket.org/ravatar/{1dc39ab6-2798-450d-af2f-e976f94b5794}ts=2009435"}}, "type": "repository", "name": "dedalus", "full_name": "dedalus-project/dedalus", "uuid": "{1dc39ab6-2798-450d-af2f-e976f94b5794}"}, "links": {"attachments": {"href": "data/repositories/dedalus-project/dedalus/issues/62/attachments_page=1.json"}, "self": {"href": "data/repositories/dedalus-project/dedalus/issues/62.json"}, "watch": {"href": "https://api.bitbucket.org/2.0/repositories/dedalus-project/dedalus/issues/62/watch"}, "comments": {"href": "data/repositories/dedalus-project/dedalus/issues/62/comments_page=1.json"}, "html": {"href": "#!/dedalus-project/dedalus/issues/62/installing-pyqt5"}, "vote": {"href": "https://api.bitbucket.org/2.0/repositories/dedalus-project/dedalus/issues/62/vote"}}, "reporter": null, "title": "Installing PyQt5", "component": null, "votes": 0, "watches": null, "content": {"raw": "I may have stumbled upon a minor bug, though I'm still not sure.\r\n\r\nI just did a fresh install of dedalus on my mac using the conda method, and when I tried to run something, the PyQt5 module was missing. The full error message was:\r\n\r\n\r\n```\r\n#!bash\r\n\r\n    Traceback (most recent call last):\r\n      File \"NGsimple.py\", line 43, in <module>\r\n        from dedalus.extras.plot_tools import plot_bot_2d\r\n      File \"/Users/NicoG/anaconda/envs/dedalus/lib/python3.7/site-packages/dedalus/extras/plot_tools.py\", line 5, in <module>\r\n        import matplotlib.pyplot as plt\r\n      File \"/Users/NicoG/anaconda/envs/dedalus/lib/python3.7/site-packages/matplotlib/pyplot.py\", line 2349, in <module>\r\n        switch_backend(rcParams[\"backend\"])\r\n      File \"/Users/NicoG/anaconda/envs/dedalus/lib/python3.7/site-packages/matplotlib/pyplot.py\", line 215, in switch_backend\r\n        backend_mod = importlib.import_module(backend_name)\r\n      File \"/Users/NicoG/anaconda/envs/dedalus/lib/python3.7/importlib/__init__.py\", line 127, in import_module\r\n        return _bootstrap._gcd_import(name[level:], package, level)\r\n      File \"/Users/NicoG/anaconda/envs/dedalus/lib/python3.7/site-packages/matplotlib/backends/backend_qt5agg.py\", line 11, in <module>\r\n        from .backend_qt5 import (\r\n      File \"/Users/NicoG/anaconda/envs/dedalus/lib/python3.7/site-packages/matplotlib/backends/backend_qt5.py\", line 15, in <module>\r\n        import matplotlib.backends.qt_editor.figureoptions as figureoptions\r\n      File \"/Users/NicoG/anaconda/envs/dedalus/lib/python3.7/site-packages/matplotlib/backends/qt_editor/figureoptions.py\", line 13, in <module>\r\n        from matplotlib.backends.qt_compat import QtGui\r\n      File \"/Users/NicoG/anaconda/envs/dedalus/lib/python3.7/site-packages/matplotlib/backends/qt_compat.py\", line 158, in <module>\r\n        raise ImportError(\"Failed to import any qt binding\")\r\n    ImportError: Failed to import any qt binding\r\n```\r\n\r\n\r\nGoogle it and you will find all over that PyQt5 needs to be installed. This solved it for me:\r\n\r\npip install PyQt5\r\n\r\nwhich you may want to add to the install_conda.sh.\r\n\r\nBefore that, if I opened an ipython shell in the dedalus environment and tried `import PyQt5`, there was no error message. Confused me for a bit but the solution above did the trick anyway.", "markup": "markdown", "html": "<p>I may have stumbled upon a minor bug, though I'm still not sure.</p>\n<p>I just did a fresh install of dedalus on my mac using the conda method, and when I tried to run something, the PyQt5 module was missing. The full error message was:</p>\n<div class=\"codehilite language-bash\"><pre><span></span>    Traceback <span class=\"o\">(</span>most recent call last<span class=\"o\">)</span>:\n      File <span class=\"s2\">&quot;NGsimple.py&quot;</span>, line <span class=\"m\">43</span>, in &lt;module&gt;\n        from dedalus.extras.plot_tools import plot_bot_2d\n      File <span class=\"s2\">&quot;/Users/NicoG/anaconda/envs/dedalus/lib/python3.7/site-packages/dedalus/extras/plot_tools.py&quot;</span>, line <span class=\"m\">5</span>, in &lt;module&gt;\n        import matplotlib.pyplot as plt\n      File <span class=\"s2\">&quot;/Users/NicoG/anaconda/envs/dedalus/lib/python3.7/site-packages/matplotlib/pyplot.py&quot;</span>, line <span class=\"m\">2349</span>, in &lt;module&gt;\n        switch_backend<span class=\"o\">(</span>rcParams<span class=\"o\">[</span><span class=\"s2\">&quot;backend&quot;</span><span class=\"o\">])</span>\n      File <span class=\"s2\">&quot;/Users/NicoG/anaconda/envs/dedalus/lib/python3.7/site-packages/matplotlib/pyplot.py&quot;</span>, line <span class=\"m\">215</span>, in switch_backend\n        <span class=\"nv\">backend_mod</span> <span class=\"o\">=</span> importlib.import_module<span class=\"o\">(</span>backend_name<span class=\"o\">)</span>\n      File <span class=\"s2\">&quot;/Users/NicoG/anaconda/envs/dedalus/lib/python3.7/importlib/__init__.py&quot;</span>, line <span class=\"m\">127</span>, in import_module\n        <span class=\"k\">return</span> _bootstrap._gcd_import<span class=\"o\">(</span>name<span class=\"o\">[</span>level:<span class=\"o\">]</span>, package, level<span class=\"o\">)</span>\n      File <span class=\"s2\">&quot;/Users/NicoG/anaconda/envs/dedalus/lib/python3.7/site-packages/matplotlib/backends/backend_qt5agg.py&quot;</span>, line <span class=\"m\">11</span>, in &lt;module&gt;\n        from .backend_qt5 import <span class=\"o\">(</span>\n      File <span class=\"s2\">&quot;/Users/NicoG/anaconda/envs/dedalus/lib/python3.7/site-packages/matplotlib/backends/backend_qt5.py&quot;</span>, line <span class=\"m\">15</span>, in &lt;module&gt;\n        import matplotlib.backends.qt_editor.figureoptions as figureoptions\n      File <span class=\"s2\">&quot;/Users/NicoG/anaconda/envs/dedalus/lib/python3.7/site-packages/matplotlib/backends/qt_editor/figureoptions.py&quot;</span>, line <span class=\"m\">13</span>, in &lt;module&gt;\n        from matplotlib.backends.qt_compat import QtGui\n      File <span class=\"s2\">&quot;/Users/NicoG/anaconda/envs/dedalus/lib/python3.7/site-packages/matplotlib/backends/qt_compat.py&quot;</span>, line <span class=\"m\">158</span>, in &lt;module&gt;\n        raise ImportError<span class=\"o\">(</span><span class=\"s2\">&quot;Failed to import any qt binding&quot;</span><span class=\"o\">)</span>\n    ImportError: Failed to import any qt binding\n</pre></div>\n\n\n<p>Google it and you will find all over that PyQt5 needs to be installed. This solved it for me:</p>\n<p>pip install PyQt5</p>\n<p>which you may want to add to the install_conda.sh.</p>\n<p>Before that, if I opened an ipython shell in the dedalus environment and tried <code>import PyQt5</code>, there was no error message. Confused me for a bit but the solution above did the trick anyway.</p>", "type": "rendered"}, "assignee": null, "state": "wontfix", "version": null, "edited_on": null, "created_on": "2019-06-11T22:24:17.727363+00:00", "milestone": null, "updated_on": "2020-03-19T18:52:25.859987+00:00", "type": "issue", "id": 62}